gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/metrics/kafka/KafkaAvroSchemaRegistry.java

```diff
@@ -85,7 +85,18 @@ public KafkaAvroSchemaRegistry(Properties props) {
     GenericObjectPoolConfig config = new GenericObjectPoolConfig();
     config.setMaxTotal(objPoolSize);
     config.setMaxIdle(objPoolSize);
-    this.httpClientPool = new GenericObjectPool<>(new HttpClientFactory(), config);
+
+    HttpClientFactory factory;
+    if (this.props.containsKey(ConfigurationKeys.KAFKA_SCHEMA_REGISTRY_HTTPCLIENT_SO_TIMEOUT)
```

Comment:
If you use -1 as default then this can be simplified and allows setting the timeouts independently.